source/nodejs/adaptivecards/src/card-elements.ts

```diff
@@ -2953,6 +2953,208 @@ export class TextInput extends Input {
+  static readonly styleProperty = new EnumProperty(Versions.v1_0, "style", Enums.InputTextStyle, Enums.InputTextStyle.File);
+  static readonly regexProperty = new StringProperty(Versions.v1_3, "regex", true);
+
+  @property(FileInput.valueProperty)
+  defaultValue?: FileResult;
+
+  @property(FileInput.maxLengthProperty)
+  maxLength?: number;
+
+  @property(FileInput.placeholderProperty)
+  placeholder?: string;
+
+  @property(FileInput.styleProperty)
+  style: Enums.InputTextStyle = Enums.InputTextStyle.File;
+
+  @property(FileInput.regexProperty)
+  regex?: string;
+
+  //#endregion
+
+  private setupInput(input: HTMLInputElement) {
+      input.style.flex = "1 1 auto";
+      input.tabIndex = 0;
+
+      if (this.placeholder) {
+          input.placeholder = this.placeholder;
+          input.setAttribute("aria-label", this.placeholder)
+      }
+
+      if (this.maxLength && this.maxLength > 0) {
+          input.maxLength = this.maxLength;
+      }
+
+      input.oninput = () => {
+          this.valueChanged();
+          let fileName = (<HTMLInputElement>this.renderedInputControlElement).value;
+          let file = (<HTMLInputElement>this.renderedInputControlElement).files?.item(0) as File;
+          let fileReader = new FileReader();
+          fileReader.onload = () => {
+              let content = fileReader.result as string;
+              this.defaultValue = new FileResult(fileName, content);
+          }
+          fileReader.readAsDataURL(file);
+      }
+  }
+
+  protected internalRender(): HTMLElement | undefined {
+      let result: HTMLInputElement;
+
+      result = document.createElement("input");
+      result.className = this.hostConfig.makeCssClassName("ac-input", "ac-textInput");
+      result.type = Enums.InputTextStyle[this.style].toLowerCase();
+
+      this.setupInput(result);
+
+      return result;
+  }
+
+  protected overrideInternalRender(): HTMLElement | undefined {
+      let renderedInputControl = super.overrideInternalRender();
+      return renderedInputControl;
+  }
+
+  getJsonTypeName(): string {
+      return "Input.File";
+  }
+
+  getActionById(id: string) {
+      let result = super.getActionById(id);
+      return result;
+  }
+
+  isSet(): boolean {
+      return this.value ? true : false;
+  }
+
+  get value(): string | undefined {
+      if (this.defaultValue) {
+          return JSON.stringify(this.defaultValue, undefined, 4);
+      }
+      else {
+          return undefined;
+      }
+  }
+}
+
+export class ColorInput extends Input {
+  //#region Schema
+
+  static readonly valueProperty = new StringProperty(Versions.v1_0, "value");
+  static readonly maxLengthProperty = new NumProperty(Versions.v1_0, "maxLength");
+  static readonly placeholderProperty = new StringProperty(Versions.v1_0, "placeholder");
+  static readonly styleProperty = new EnumProperty(Versions.v1_0, "style", Enums.InputTextStyle, Enums.InputTextStyle.Color);
```

Comment:
`Enums.InputTextStyle` is used to specify the style of the otherwise generic Input.Text.

If you're going to add new File and Color styles to the enum, these will be available on TextInput itself, but it won't really work. So you have two choices:
- Either you do add the enum values but then:
  - You have to make TextInput honor them (and thus you have to update TextInput itself to implement the logic you currently have in your FileInput and ColorInput classes)
  - And thus you remove you FileInput and ColorInput classes
- Or:
  - You don't add the enum values
  - You don't declare a style property on either FileInput or ColorInput because it's no necessary (the classes themselves represent specialized inputs, no need to have a style property that repeats that)

To me there's no doubt the second option is the right one.